enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,351 @@
+
+The telco users have new use cases and requirements that especially related to resource management of the platform. 
+In order to address some of those needs,
+Kubelet CPU manager was created to allow users requesting for exclusive, isolated cpus.
+
+But there are additional use cases that require more granular control of CPU management.
+This enhancement aims to solve the following use case:
+When the CPU manager allocates an exclusive set of CPUs for a given workloads, the workload shall still have the option 
+to specify and access a set of shared CPUs, whether those shared CPUs are shared among containers in the 
+same pod or containers in different ones.
+
+By giving the users the flexibility to use exclusive CPUs only when it is an absolute necessity, we help them utilize
+their existing compute resources in an optimal manner.
+
+### User Stories
+
+* As a developer who's building DPDK application,
+I want to run my application in a container,
+giving full exclusive access to cpu cores for the busy-loops polling threads,
+but having the light-weight tasks like configuration processing and log printing not taking full cores,
+in order to improve the application density and make more efficient use of the CPUs.
+For that, my application container needs access to both shared and exclusive CPUs.
+
+  
+### Goals 
+* Increase pod density, power efficiency, cost reduction and sustainability by optimizing CPUs resource management. 
+* Provide a special-purpose at a runtime-level solution for containers in a pod to have the ability
+to request and subsequently be allocated both exclusive and shared CPUs.
+* Enable the telco-specific use case with a self-contained change,
+    minimizing the changes to the platform
+
+### Non-Goals
+* Introducing a generic mechanism in the platform that does involve Kubelet and pod spec changes.
+
+## Proposal
+
+Container processes are restricted
+to given [cpuset](https://man7.org/linux/man-pages/man7/cpuset.7.html) by their corresponding cgroup definition.
+Essentially, the kernel enforces the processes of container to run only at the CPU (core) ids specified under its
+cgroup.
+
+The mixed-cpu-node-plugin is a  runtime-agnostic, hybrid (composed of both [NRI-plugin](https://github.com/containerd/nri) and
+[device-plugin](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/)), 
+[mixed-cpu-node-plugin](https://github.com/openshift-kni/mixed-cpu-node-plugin) 
+that communicates with the container-runtime
+at various lifecycle stages of container e.g. CreateContainer, UpdateContainer, etc.,
+and modifies the container's cgroup to include a set of shared CPUs.
+
+At first,
+shared CPUs will be part of Kubelet [reservedSystemCpus](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#explicitly-reserved-cpu-list).
+The reason to utilize Kubelet `reservedSystemCpus` is because of CPU manager, which is not aware of the CPUs
```

Comment:
There is risk of system being starved of cpus here.

Reply:
Thank you @mrunalp for taking the time to look into this EP.
The user shall specify additional cpus under the performanceProfile `spec.cpu.shared` filed and those cpus appends to Kubelet `reservedSystemCpus`.

IOW, when the feature enabled, Kubelet `reservedSystemCpus` is composed of performanceProfile `spec.cpu.reserved` + `spec.cpu.shared`.

Moreover, the plugin cannot assign the cpus under the `spec.cpu.reserved`, so the amount of cpus for OCP infra remains the same.

Reply:
I updated the EP to reflect this information better.